tidal/tidal.go

```diff
@@ -128,6 +128,18 @@ func (c *Client) TryInitiateLoginFlow(
 	return link, wait, nil
 }
 
+// NormalizePathParts normalizes URL path parts by handling "browse" prefix and "u" suffix.
+// This is used to support both old and new TIDAL link formats.
+func NormalizePathParts(pathParts []string) []string {
+	if len(pathParts) >= 1 && pathParts[0] == "browse" {
+		pathParts = pathParts[1:]
+	} else if len(pathParts) >= 3 && pathParts[2] == "u" {
+		pathParts = pathParts[:2]
+	}
+
+	return pathParts
+}
+
```

Comment:
Make it so that it doesn't assign the output to the parameter, and it both removes the `"browse"` amd `"u"` segments.